canonical-web-and-design/vanilla-framework
# Add side navigation component with default styling

## Done

- adds basic version of side navigation
- adds examples and simple docs

Fixes #2860 
Fixes #2861

## QA

- Pull code
- Run `./run`
- Open http://0.0.0.0:8101/ or [demo](https://vanilla-framework-canonical-web-and-design-pr-2915.run.demo.haus/docs/examples/patterns/side-navigation/default)
- Check new side navigation pattern examples, make sure they work correctly
  - [default](https://vanilla-framework-canonical-web-and-design-pr-2915.run.demo.haus/docs/examples/patterns/side-navigation/default)
  - [docs](https://vanilla-framework-canonical-web-and-design-pr-2915.run.demo.haus/docs/examples/patterns/side-navigation/docs)
- make sure side nav documentation is added: https://vanilla-framework-canonical-web-and-design-pr-2915.run.demo.haus/docs/patterns/navigation/#side-navigation
  - NOTE: codepen will not render the styles for sidenav, because it uses published version of Vanilla


## Screenshots

<img width="323" alt="Screenshot 2020-03-12 at 10 13 11" src="https://user-images.githubusercontent.com/83575/76505955-a2359700-644a-11ea-9838-74c1c6fa785b.png">
